### ISSUE TITLE
Recognize full Unicode range document slugs

### ISSUE THREAD

**Issue by reefdog** [documentcloud/wordpress-documentcloud#37]
Currently, the platform can create document slugs with Unicode characters, but our pattern-recognizer still only recognizes Latin alphanumerics.

Need to change [`0-9a-zA-Z-`](https://github.com/documentcloud/wordpress-documentcloud/blob/4de1d222b4ce517b71083a8d4c62a5c3401760f3/documentcloud.php#L35) to `\p{L}\p{N}%-`.

**However!** We're blocked by the platform's oEmbed endpoint, which currently chokes on them.

**Comment by reefdog** [comment]
Meh. We should be in sync with the platform. That it won't help it work *now* is immaterial; when the oEmbed endpoint *does* work properly, it'll be nice to have the plugin already ready. Hotfixing this as v0.4.2.
